### PR TITLE
T01 completed

### PR DESCRIPTION
**Pull request by YoganathanM**

### CODE REVIEW

File: Assignment/Program.cs [hunk 1]
```diff
@@ -1,7 +1,49 @@
-﻿namespace Assignment {
-   internal class Program {
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello world!");
+﻿for (; ; ) {
+   Console.Write ("Enter a number : ");
+   string decimalNumber = Console.ReadLine ();
```
Comment (varalakshmi-imm) [inline Assignment/Program.cs:3]: decimalNumber not a good name. Could be _string s_ or _string str_

File: Assignment/Program.cs [hunk 2]
```diff
@@ -1,7 +1,49 @@
-﻿namespace Assignment {
-   internal class Program {
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello world!");
+﻿for (; ; ) {
+   Console.Write ("Enter a number : ");
+   string decimalNumber = Console.ReadLine ();
+   if (int.TryParse (decimalNumber, out int decimalNum)) {
+      //Decimal to Binary
```
Comment (varalakshmi-imm) [inline Assignment/Program.cs:5]: Always leave a space after the slashes in comment
// Decimal to Binary

File: Assignment/Program.cs [hunk 3]
```diff
@@ -1,7 +1,49 @@
...
-   internal class Program {
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello world!");
+﻿for (; ; ) {
+   Console.Write ("Enter a number : ");
+   string decimalNumber = Console.ReadLine ();
+   if (int.TryParse (decimalNumber, out int decimalNum)) {
+      //Decimal to Binary
+      int dec = decimalNum, i = 0;
+      var l = new List<int> ();
```
Comment (varalakshmi-imm) [inline Assignment/Program.cs:7]: Instead of list this could use a stringbuilder

File: Assignment/Program.cs [hunk 4]
```diff
@@ -1,7 +1,49 @@
...
+   { 10, 'A'},
+   {11, 'B' },
+   {12, 'C' },
+   {13, 'D' },
+   {14, 'E' },
+   {15, 'F' }
+};
+      var a = new List<int> ();
+      while (dec2 > 0) {
+         int rem = dec2 % 16;
```
Comment (varalakshmi-imm) [inline Assignment/Program.cs:32]: Some code repeats for binary and hex. So you can refactor and simplify both the conversions in a single method which can take parameters to say if you want binary or hex.

**Review by varalakshmi-imm — CHANGES_REQUESTED** [review]

File: Assignment/Program.cs (hunk 1 above)
Comment (YoganathanM) [reply Assignment/Program.cs:3]: ok

File: Assignment/Program.cs (hunk 2 above)
Comment (YoganathanM) [reply Assignment/Program.cs:5]: ok

File: Assignment/Program.cs (hunk 4 above)
Comment (YoganathanM) [reply Assignment/Program.cs:32]: ok

File: Assignment/Program.cs (hunk 3 above)
Comment (YoganathanM) [reply Assignment/Program.cs:7]: ok

File: Assignment/Program.cs [hunk 5]
```diff
@@ -1,7 +1,36 @@
...
+   while (num > 0) {
+      sb.Insert (0, num % 2);
+      num /= 2;
    }
+   Console.WriteLine ($"Binary : {sb}");
+}
+
+static void ToHexaDecimal (int num) {
+   Dictionary<int, char> dict = new ()
+   {
```
Comment (varalakshmi-imm) [inline Assignment/Program.cs:23]: This open bracket should be in the line above next to new ()

File: Assignment/Program.cs [hunk 6]
```diff
@@ -1,7 +1,36 @@
...
+static void ToBinary (int num) {
+   var sb = new StringBuilder ();
+   while (num > 0) {
+      sb.Insert (0, num % 2);
+      num /= 2;
    }
+   Console.WriteLine ($"Binary : {sb}");
+}
+
+static void ToHexaDecimal (int num) {
```
Comment (varalakshmi-imm) [inline Assignment/Program.cs:21]: Method name could be just ToHex.

**Review by varalakshmi-imm — CHANGES_REQUESTED** [review]
This implementation is fine. But also look at [this](https://learn.microsoft.com/en-us/dotnet/api/system.convert.tostring?view=net-7.0#system-convert-tostring(system-int32-system-int32)).
Convert.ToString with a base value does the job required.
Also for hex _num.ToString("X")_ this interpolation will also work.
You can extend the program to use this to verify your computed value using ToBinary or ToHex methods.

File: Assignment/Program.cs [hunk 7]
```diff
@@ -1,26 +1,27 @@
...
-   Console.Write ("Enter a number : ");
-   string input = Console.ReadLine ();
-   if (int.TryParse (input, out int num)) {
-      ToBinary (num);
-      ToHexaDecimal (num);
+using static System.Console;
+while (true) {
+   Write ("Enter a number : ");
+   if (int.TryParse (ReadLine (), out int num) && num >= 0) {
+      if (Convert (num, 2) == ToBinary (num) && Convert (num, 16) == ToHexa (num)) {
```
Comment (varalakshmi-imm) [inline Assignment/Program.cs:6]: (Convert (num, 2) and (Convert (num, 16) called twice.
Also the check is not needed.
Just print them maybe like Binary computation and Binary string conversion etc.

**Review by varalakshmi-imm — CHANGES_REQUESTED** [review]

**Review by varalakshmi-imm — APPROVED** [review]

**Pull request closed without merging**